x/sync/network_client.go

```diff
@@ -250,9 +246,18 @@ func (c *networkClient) request(
 	handler := newResponseHandler()
 	c.outstandingRequestHandlers[requestID] = handler
 
+	var (
+		response  []byte
+		startTime = time.Now()
+	)
+	defer func() {
+		elapsed := time.Since(startTime).Seconds()
+		bandwidth := float64(len(response))/elapsed + epsilon
+		c.peers.TrackBandwidth(nodeID, bandwidth)
+	}()
```

Comment:
I don't think this `defer` honors the prior semantics - If `handler.failed` then we should be `c.peers.TrackBandwidth(nodeID, 0)` right?

Reply:
If `handler.failed` then `len(response)` so `bandwidth` is `0`

Reply:
`+ epsilon`